File: back/models/Teacher.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const teacherSchema = new mongoose.Schema({
  user: {
    type: Schema.Types.ObjectId,
    ref: "User",
    required: true,
  },
  firstname: {
    type: String,
    required: true,
  },
  lastname: {
    type: String,
    required: true,
  },
  age: {
    type: Number,
    required: true,
  },
  promotions: [
    {
      type: Schema.Types.ObjectId,
      ref: "Promotion",
      required: false,
    },
  ],
  lessons: [
    {
      type: Schema.Types.ObjectId,
      ref: "Lesson",
      required: false,
    },
  ],
});

module.exports = mongoose.model("Teacher", teacherSchema);

File: h1/lib/models/teacher.dart
import 'package:hive/hive.dart';

part 'teacher.g.dart';

@HiveType(typeId: 2)
class Teacher {
  @HiveField(0)
  final String firstname;

  @HiveField(1)
  final String lastname;

  Teacher({
    required this.firstname,
    required this.lastname,
  });
}